Add util types

devanshutanna opened the pull request:
## Overview

- Description: Add types and test for the old and new types
- Schema update type: extend
- Services affected: none

## How was this tested?

Yes, have created type tests..

KillDozerX2 reviewed (changes requested):
Please make changes as requested in the call.

devanshutanna committed:
feat: add types for creating model from openApi path

devanshutanna committed:
chore(deps): update to new ts 4.9.5

devanshutanna committed:
test: add test for utils

devanshutanna committed:
test: add test for from param

devanshutanna committed:
test: add test for from operation

devanshutanna committed:
feat(config-helpers): add config helpers

for combining OpenAPI paths, getting OpenAPI PathsObject and for AWS Handler definition

devanshutanna committed:
test(config-helpers): add tests for config helpers

KillDozerX2 commented on package.json:
```diff
@@ -17,6 +17,9 @@
   "files": [
     "lib/**/*"
   ],
+  "engines": {
+    "node": ">=12"
+  },
```
@devanshutanna Let's bump this to >= 16.

devanshutanna replied:
Ok 👍

KillDozerX2 commented on package.json:
```diff
@@ -70,7 +73,7 @@
     "serverless-step-functions": "^3.12.1",
     "ts-node": "^10.9.1",
-    "typescript": "^4.9.0-dev.20220916"
+    "typescript": "^4.9.5"
```
@devanshutanna Is this for `satisfies`?

devanshutanna replied:
We did have support for satisfies already..
Just that we were using a dev build,
So updated it to the latest version.

KillDozerX2 replied:
Understood, thanks

KillDozerX2 reviewed (changes requested):
Good to go, just waiting for docs and a few clarifications.
Also, you might have to rebase.

devanshutanna committed:
docs(config-helper): add documentation for config helpers

devanshutanna committed:
chore: use node engine >=16
